lib/core/facets.rb

```diff
@@ -1,3 +1,10 @@
+unless $FACETS_LOADED
```

Comment:
Just reading this the first time, the naming confused me.

Perhaps `$IGNORE_FACETS_WARNING` or something else would be more descriptive?

I think the global variable could be replaced with something like this:

```ruby
module Ecraft
  module Facets
    class << self
      attr_reader :already_warned
    end
  end
end

unless Ecraft::Facets.already_warned
  # deprecation code
end